src/ducks/AlternativeStore/transformData.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  AltStoreConfig,
+  AltStoreSourceShortcut
+} from 'ducks/AlternativeStore/types'
+
+export const transformData = (
+  data: AltStoreSourceShortcut[],
+  config: AltStoreConfig
+): AltStoreSourceShortcut[] => {
+  const categoryPathMap: { [key: string]: string } = {}
+  for (const [key, path] of Object.entries(config.categories)) {
+    categoryPathMap[path] = key
+  }
+
+  const fileTypeMappings: { [key: string]: string } = {}
+  for (const [fileType, categoryName] of Object.entries(
+    config.fileTypeMappings
+  )) {
+    fileTypeMappings[fileType] = categoryName
+  }
+
+  return data.map(file => {
+    const metadataType = file.metadata.type || 'default' // Handle undefined type
+    const filePath = file.path
+
+    if (!filePath) return file
```

Comment:
By default file don't have a path. From what I understood, you keep all shortcut even if they are not in a folder from the config 🤔